runtime/lua/vim/lsp/buf.lua

```diff
@@ -192,17 +192,21 @@ end
 ---     - range (table|nil) Range to format.
 ---         Table must contain `start` and `end` keys with {row,col} tuples using
 ---         (1,0) indexing.
+---         Can also be a list of tables that contain `start` and `end` keys as described above,
+---         in which case textDocument/rangesFormatting support is required.
```

Comment:
What kind of error is presented to the user if the language client doesn't support `ranges`? Would it make sense to check for the capability here for a better error message?

Reply:
The call to get_client() already filters out any servers that do not support the LSP method, so you'd get `[LSP] Format request failed, no matching language servers.` if none of the servers support `textDocument/rangesFormatting` (the same thing happens for `textDocument/rangeFormatting` alrady

Reply:
Sounds good :)